src/Shipping.php

```diff
@@ -0,0 +1,253 @@
+	 * Normally this would be done by checking values in cart. Because we are not operating on cart we filter out
+	 * this value ourselves.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param bool                      $is_available    Wether this shipping method should be available.
+	 * @param array                     $package         Shipping package.
+	 * @param WC_Shipping_Free_Shipping $shipping_method Shipping method.
+	 * @return boolean
+	 */
+	public static function is_free_shipping_available( $is_available, $package, $shipping_method ) {
+		if ( $is_available ) {
+			return $is_available;
+		}
+
+		if ( ! in_array( $shipping_method->requires, array( 'min_amount' ), true ) ) {
+			return $is_available;
+		}
+
+		$has_met_min_amount = $package['cart_subtotal'] >= $shipping_method->min_amount;
+
+		return $has_met_min_amount;
+	}
+
+	/**
+	 * Get the lowest possible shipping cost for given location and shipping methods for that location.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param array      $shipping_location Country and state values of the location.
+	 * @param array      $shipping_methods  List of shippings methods we use to calculate the best rate.
+	 * @param WC_Product $product           Product for which we want to generate the shipping column.
+	 * @return array|null                   Name and cost for the best found rate or null in case nothing was found.
+	 */
+	private static function get_best_shipping_with_cost( $shipping_location, $shipping_methods, $product ) {
+
+		// Since in a shipping zone all locations are treated the same we will perform the calculations for the first one.
+		$package = self::put_product_into_a_shipping_package( $product, $shipping_location );
+		$rates   = array();
+
+		// Substitute default customer location and billing for.
+		WC()->customer = new \WC_Customer( get_current_user_id() );
+		WC()->customer->set_billing_location( $shipping_location['country'], $shipping_location['state'] );
+		WC()->customer->set_shipping_location( $shipping_location['country'], $shipping_location['state'] );
+
+		// By using the filter we can trick the get_rates_for_package to continue calculations even without having the Cart defined.
+		add_filter( 'woocommerce_shipping_free_shipping_is_available', array( static::class, 'is_free_shipping_available' ), 10, 3 );
+		foreach ( $shipping_methods as $shipping_method ) {
+			$rates += $shipping_method->get_rates_for_package( $package );
+		}
+		remove_filter( 'woocommerce_shipping_free_shipping_is_available', array( static::class, 'is_free_shipping_available' ), 10 );
+
+		// Check if shipping methods have returned any valid rates.
+		if ( empty( $rates ) ) {
+			return null;
+		}
+
+		$best_rate = self::calculate_best_rate( $rates );
+		return $best_rate;
+	}
+
+	/**
+	 * Pick the best rate from an array of rates.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param  array $rates List of shipping rates.
+	 * @return array        Name and cost for the best found rate or null in case nothing was found.
```

Comment:
❓ 

In case nothing was found it would return an empty name and a cost of  "INF",  should it not return null?

Reply:
Corrected.